Makefile

```diff
@@ -1,35 +1,45 @@
-.PHONY: all
-all: check
+GO111MODULE ?= on
+export GO111MODULE
+GOFLAGS ?= -mod=vendor
+export GOFLAGS
+GOPROXY ?=
+export GOPROXY
 
 NO_DOCKER ?= 0
 ifeq ($(NO_DOCKER), 1)
   DOCKER_CMD =
   IMAGE_BUILD_CMD = imagebuilder
-  CGO_ENABLED = 1
 else
-  DOCKER_CMD := docker run --rm -v "$(PWD)":/go/src/github.com/openshift/cluster-api-actuator-pkg:Z -w /go/src/github.com/openshift/cluster-api-actuator-pkg openshift/origin-release:golang-1.12
+  DOCKER_CMD := docker run \
+	  --rm \
+	  -v "$(PWD)":/go/src/github.com/openshift/cluster-api-actuator-pkg:Z \
+	  -w /go/src/github.com/openshift/cluster-api-actuator-pkg \
+	  -e GO111MODULE=$(GO111MODULE) \
+	  -e GOFLAGS=$(GOFLAGS) \
+	  -e GOPROXY=$(GOPROXY) \
+	  openshift/origin-release:golang-1.12
   IMAGE_BUILD_CMD = docker build
 endif
 
-.PHONY: depend
-depend:
-	dep version || go get -u github.com/golang/dep/cmd/dep
-	dep ensure
+.PHONY: all
+all: check
 
-.PHONY: depend-update
-depend-update:
-	dep ensure -update
+.PHONY: vendor
+vendor:
+	go mod tidy
+	go mod vendor
+	go mod verify
 
 .PHONY: check
 check: fmt vet lint test ## Check your code
 
 .PHONY: test
 test: # Run unit test
-	$(DOCKER_CMD) go test -race -cover `go list ./... | grep -v github.com/openshift/cluster-api-actuator-pkg/pkg/e2e`
+	$(DOCKER_CMD) go test -race -cover $$(go list ./... | grep -v github.com/openshift/cluster-api-actuator-pkg/pkg/e2e)
 
 .PHONY: lint
 lint: ## Go lint your code
-	hack/go-lint.sh -min_confidence 0.3 $(go list -f '{{ .ImportPath }}' ./...)
+	hack/go-lint.sh -min_confidence 0.3 $(go list ./...)
```

Comment:
I think this should be double `$$` or: `$(shell go list ./...)`

Also, I don't understand why we duplicate the Docker stuff in the hack scripts, but that's not related to this.